src/Http/Http.Extensions/gen/StaticRouteHandlerModel/StaticRouteHandlerModel.Emitter.cs

```diff
@@ -161,6 +163,53 @@ public static void EmitFilteredRequestHandler(this Endpoint endpoint, CodeWriter
         codeWriter.EndBlock(); // End handler method block
     }
 
+    private static void EmitBuiltinResponseTypeMetadata(this Endpoint endpoint, CodeWriter codeWriter)
+    {
+        if (endpoint.Response is not { } response || response.ResponseType is not { } responseType)
+        {
+            return;
+        }
+
+        if (response.HasNoResponse || response.IsIResult)
+        {
+            codeWriter.WriteLine("// No built-in IProducesResponseTypeMetadata is generated because response is void or is IResult.");
+            return;
+        }
+
+        if (responseType.SpecialType == SpecialType.System_String)
+        {
+            codeWriter.WriteLine($$"""options.EndpointBuilder.Metadata.Add(new GeneratedProducesResponseTypeMetadata(type: null, statusCode: 200, contentTypes: new string[] { "text/plain" }));""");
```

Comment:
Nit:
```suggestion
            codeWriter.WriteLine($$"""options.EndpointBuilder.Metadata.Add(new GeneratedProducesResponseTypeMetadata(type: null, statusCode: StatusCodes.Status200OK, contentTypes: new string[] { "text/plain" }));""");
```

Reply:
`new string[] { "text/plain" }` should be cached somewhere.